### DISCUSSION TITLE
How to adjust the Serviceability Limits in SteelDesignServiceabilityConfigurations

### DISCUSSION THREAD

**Discussion by MFE-VIRO** [Dlubal-Software/RFEM_Python_Client#289] [Q&A] [answered]
I would like to know if it is possible to adjust the serviceability limit settings with Python. I couldn't find any parameters for SteelDesignServiceabilityConfigurations that would allow you to do this. That's why I made a simple model in RFEM6 in which I adjusted the characteristic limit: 

![2023-10-01 09_55_00-Edit Serviceability Configuration _ Steel Design _ EN 1993 _ CEN _ 2015-06](https://github.com/Dlubal-Software/RFEM_Python_Client/assets/131137450/e950e9bb-4da1-4786-ac4a-901a42e92ce2)

I then used the WSscripGenerator example to create a script from this model, hoping that I would be able to find the appropriate parameter in the script. Unfortunately, no parameter appeared for this at SteelDesignServiceabilityConfigurations:

`SteelDesignServiceabilityConfigurations(params={'no': 1, 'user_defined_name_enabled': True, 'name': 'EC3 checks BGT', 'assigned_to_all_members': True, 'assigned_to_members': '1', 'assigned_to_all_member_sets': True})`

In fact, when I run the script, it creates a model with the steel addon not even enabled...

See [QuestionServiceabilityLimit.zip](https://github.com/Dlubal-Software/RFEM_Python_Client/files/12776708/QuestionServiceabilityLimit.zip).

What am I doing wrong? Am I missing something?
Can you give me some feedback on this?
Thank you in advance!

Kind regards, Martin Francke

**Comment by heetrojivadiya** [comment, marked as answer]
Hello @MFE-VIRO ,

Thank you for reaching us. At the moment, we can not excess steel design sls with python. Although, you can set steel design sls with javaScript. So, you can create temporary .js file and which you can run in python with help of run_script() function. Here, I have attached small example of it. 

```py
import os
baseName = os.path.basename(__file__)
dirName = os.path.dirname(__file__)

from RFEM.enums import *
from RFEM.initModel import Model, SetAddonStatus
from RFEM.BasicObjects.material import Material
from RFEM.BasicObjects.member import Member
from RFEM.BasicObjects.section import Section
from RFEM.BasicObjects.node import Node
from RFEM.SteelDesign.steelServiceabilityConfiguration import SteelDesignServiceabilityConfigurations

def BeamLimitCharacteristic(value: int = 100, sls_no = 1):

    with open(dirName+r"./sls_char.js", "w") as std:

        std.write("STEEL_DESIGN.steel_design_sls_configurations[{}].settings_ec3.property_sl_beam_limit_characteristic = {}".format(sls_no, value))

    Model.clientModel.service.run_script(dirName+r"./sls_char.js")

    os.remove(dirName+r"./sls_char.js")

def BeamLimitFrequent(value: int = 100, sls_no = 1):

    with open(dirName+r"./sls_char.js", "w") as std:

        std.write("STEEL_DESIGN.steel_design_sls_configurations[{}].settings_ec3.property_sl_beam_limit_frequent = {}".format(sls_no, value))

    Model.clientModel.service.run_script(dirName+r"./sls_char.js")

    os.remove(dirName+r"./sls_char.js")

def BeamLimitQuasi(value: int = 100, sls_no = 1):

    with open(dirName+r"./sls_char.js", "w") as std:

        std.write("STEEL_DESIGN.steel_design_sls_configurations[{}].settings_ec3.property_sl_beam_limit_quasi_permanent = {}".format(sls_no, value))

    Model.clientModel.service.run_script(dirName+r"./sls_char.js")

    os.remove(dirName+r"./sls_char.js")

Model(True,'Demo')
SetAddonStatus(Model.clientModel, AddOn.steel_design_active)

Node()
Node(2, 5,0,0)
Material()
Section()
Member(1,1,2,0,1,1)
SteelDesignServiceabilityConfigurations()

BeamLimitCharacteristic(100, 1)
BeamLimitFrequent(150)
BeamLimitQuasi(250)
```

This is just demo example. You can just copy the functions in your file and easily you can call with value and sls_no in your model. Let me know if this solution work for you or not. 

Best regards,
Heet Rojivadiya

**Comment by MFE-VIRO** [comment]
Hello @heetrojivadiya,

Thanks for your quick response. I will try to implement the solution soon, but in any case it seems like a solution that could work well for me.

**Reply by MFE-VIRO** [reply]
I just implemented the solution and, as expected, it works perfectly!
Thank you @heetrojivadiya !

**Reply by heetrojivadiya** [reply]
Hello @MFE-VIRO ,

Thank you for feedback. I am glad that it works perfectly. 👍 

Best regards,
Heet Rojivadiya
